Presentation/src/main/java/hu/kalee/skeleton/presentation/controller/BaseController.java

```diff
@@ -0,0 +1,140 @@
+import hu.kalee.skeleton.business.model.OutputDTO;
+import hu.kalee.skeleton.presentation.model.MessageHolder;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+/**
+ * Base Controller.
+ *
+ * @author Moondancer
+ * @since 2015.12.29..
+ *
+ * @param <A> Input DTO from frontend
+ * @param <B> Input DTO for business layer usage
+ * @param <C> Output DTO from business layer
+ * @param <D> Output DTO for frontend usage
+ */
+public abstract class BaseController<A extends InputDTO & MessageHolder, B extends InputDTO, C extends OutputDTO, D extends OutputDTO> {
+    public static final String ERROR_VIEW_NAME = "redirect:/error";
+
+    private final Converter<A, B> toBusinessConverter;
+    private final Converter<C, D> toPresentationConverter;
+
+    /**
+     * Initializing dependencies.
+     *
+     * @param toBusinessConverter     Presentation to business converter for input DTO
+     * @param toPresentationConverter Business to presentation converter for output DTO
+     */
+    public BaseController(final Converter<A, B> toBusinessConverter, final Converter<C, D> toPresentationConverter) {
+        this.toBusinessConverter = toBusinessConverter;
+        this.toPresentationConverter = toPresentationConverter;
+    }
+
+    /**
+     * Common workflow.
+     *
+     * @param input              The input DTO from frontend
+     * @param result             The validation result from frontend
+     * @param redirectAttributes The redirect attributes
+     * @return The name of next view
+     */
+    final public String process(final A input,
+                                final BindingResult result,
+                                final RedirectAttributes redirectAttributes) {
+        String viewName;
+
+        if (result.hasErrors()) {
+            viewName = getCallingViewName();
+        } else {
+            try {
+                B businessInput = prepareInput(input);
+                BusinessResult businessResult = callFacade(businessInput);
+
+                switch (businessResult.getStatus()) {
+                    case OK:
+                        D serviceResult = prepareResult(input, businessResult);
+
+                        redirectAttributes.addFlashAttribute("result", serviceResult);
+                        redirectAttributes.addFlashAttribute("message", businessResult.getMessages().get("message"));
+                        viewName = getOkViewName();
+                        break;
+                    case ERROR:
+                        redirectAttributes.addFlashAttribute("message", businessResult.getMessages().get("message"));
+                        viewName = ERROR_VIEW_NAME;
+                        break;
+                    case WARNING:
+                    default:
+                        input.addMessage("warning", businessResult.getMessages().get("warning"));
```

Comment:
Maybe warnings should be redirect to CALLING_VIEW. In that case input and massage should be sent as redirect attribute.